### PR TITLE
fix: parse input value prop to format used by flatpicker

### PR DESCRIPTION
**Pull request by connoratrug**
#### Checklist
- [x] Functionality works & meets specifications
- [x] Code reviewed
- [x] Code unit/integration/system tested
- [x] User documentation updated
- [x] Clean commits
- [x] No warnings during install
- [x] Updated flow typing

### CODE REVIEW

**Comment by ConnorStroomberg** [comment]
it solved the issue but maybe it still needs some work, added back wip label

**Comment by fdlk** [comment]
On update entity page, scroll down to date time field, open the picker, select day A, then select day B and it'll jump back to day A.

**Comment by ConnorStroomberg** [comment]
> On update entity page, scroll down to date time field, open the picker, select day A, then select day B and it'll jump back to day A.

I am unable to reproduce this behaviour, can you be more specific ?

**Comment by connoratrug** [comment]
> > On update entity page, scroll down to date time field, open the picker, select day A, then select day B and it'll jump back to day A.
> 
> I am unable to reproduce this behaviour, can you be more specific ?

Able to reproduce using firefox

File: src/components/field-types/DateFieldComponent.vue [hunk 1]
```diff
@@ -107,9 +107,7 @@ export default {
         enableSeconds: this.isTimeIncluded,
         enableTime: this.isTimeIncluded,
         dateFormat: 'Y-m-d',
-        formatDate: this.isTimeIncluded ? (date) => {
-          return moment(date).format(DATA_TIME_DISPLAY)
-        } : undefined
+        formatDate: this.isTimeIncluded ? this.toExternalDataString : undefined
```
Comment (fdlk) [inline src/components/field-types/DateFieldComponent.vue]: typo: data -> date (also in rest of file)

**Pull request merged**